ocaml/xapi/xapi_sr.ml

```diff
@@ -395,7 +395,11 @@ let update_vdis ~__context ~sr db_vdis vdi_infos =
   let find_vdi db_vdi_map loc =
     if StringMap.mem loc db_vdi_map
     then fst (StringMap.find loc db_vdi_map)
```

Comment:
It's surprising that the snapshot isn't in `db_vdi_map`. I found it strange that [when we construct the `db_vdi_map`](https://github.com/xapi-project/xen-api/blob/fe3dacf19fa179dbe4aae57c1b4e83e8dfab52bc/ocaml/xapi/xapi_sr.ml#L375) we put into it `VDI.location` keys, but then [we try to find the `VDI.snapshot_of` key in the map](https://github.com/xapi-project/xen-api/blob/fe3dacf19fa179dbe4aae57c1b4e83e8dfab52bc/ocaml/xapi/xapi_sr.ml#L422). I'm not sure if these two are the same things, as `VDI.snapshot_of` is a VDI reference, but I don't know what `VDI.location` actually is. In addition, the argument `find_vdi` function is called `loc`, and not something like `vdi`/`vdi_ref`/`ref`/... So maybe we don't find the snapshotted VDI because we're looking for the wrong key?